test/powershell/Language/Operators/NullConditional.Tests.ps1

```diff
@@ -0,0 +1,127 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+Describe 'NullConditionalOperations' -tag 'CI' {
+
+    Context "Null conditional assignment operator ??=" {
+        BeforeAll {
+
+            $skipTest = -not $EnabledExperimentalFeatures.Contains('PSNullCoalescingOperators')
+
+            if ($skipTest) {
+                Write-Verbose "Test Suite Skipped. The test suite requires the experimental feature 'PSNullCoalescingOperators' to be enabled." -Verbose
+                $originalDefaultParameterValues = $PSDefaultParameterValues.Clone()
+                $PSDefaultParameterValues["it:skip"] = $true
+            } else {
+                $someGuid = New-Guid
+                $typesTests = @(
+                    @{ name = 'string'; valueToSet = 'hello' }
+                    @{ name = 'dotnetType'; valueToSet = $someGuid }
+                    @{ name = 'byte'; valueToSet = [byte]0x94 }
+                    @{ name = 'intArray'; valueToSet = 1..2 }
+                    @{ name = 'stringArray'; valueToSet = 'a'..'c' }
+                    @{ name = 'emptyArray'; valueToSet = @(1, 2, 3) }
+                )
+            }
+        }
+
+        AfterAll {
+            if ($skipTest) {
+                $global:PSDefaultParameterValues = $originalDefaultParameterValues
+            }
+        }
+
+        It 'Variable doesnot exist' {
+
+            Remove-Variable variableDoesNotExist -ErrorAction SilentlyContinue -Force
+
+            $variableDoesNotExist ??= 1
+            $variableDoesNotExist | Should -Be 1
+
+            $variableDoesNotExist ??= 2
+            $variableDoesNotExist | Should -Be 1
+        }
+
+        It 'Variable exists and is null' {
+            $variableDoesNotExist = $null
+
+            $variableDoesNotExist ??= 2
+            $variableDoesNotExist | Should -Be 2
+        }
+
+        It 'Validate types - <name> can be set' -TestCases $typesTests {
+            param ($name, $valueToSet)
+
+            $x = $null
+            $x ??= $valueToSet
+            $x | Should -Be $valueToSet
+        }
+
+        It 'Validate hashtable can be set' {
+            $x = $null
+            $x ??= @{ 1 = '1' }
+            $x.Keys | Should -Be @(1)
+        }
+
+        It 'Validate lhs is returned' {
```

Comment:
I'd add tests with cmdlet like
```powershell
$x ??= Get-Date
$x ?? Get-Date
```

Reply:
Also we could add test for the operators in chains.
$x ??= $x ?? Func1() ?? Func2()

Reply:
Added tests